targeted/json_csrf_poc_generator.js

```diff
@@ -0,0 +1,97 @@
+//csrf poc generater supporting json csrf
+//also supports multipart/form-data .
+//it will copy the results to clipboard and print them to the zap script console
+// released under the Apache v2.0 licence.
+//You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//Author : @haseebeqx
+
+function invokeWith(msg) {
+	string = "<!DOCTYPE html>\n";
+	string += "<head>\n <title>CSRF POC</title>\n</head>";
+	string += "\n<body>";
+	if(msg.getRequestHeader().getMethod()== "POST"){
+	body = msg.getRequestBody().toString();
+	body = body.trim();
+	if(isJson(body))
+		string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\" enctype=\"text/plain\">" ;
+	else{
+		if(ismultipart(msg.getRequestHeader()))
+			string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" enctype=\"multipart/form-data\" method=\"post\">";
+		else
+			string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\">";
+	}
+	if(!isJson(body)){	
+	if(ismultipart(msg.getRequestHeader())){
+		type 	=  msg.getRequestHeader().getHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_TYPE);
+		delim 	=  type.substring(type.search("=")+1,type.length());
+		h = body.split("--"+delim);
+		k=0;
+		names = [];
+		values = [];
+		for(i =1 ; i<h.length-1;i++){
+			j = h[i].split(msg.getRequestHeader().getLineDelimiter());
+			nameField =j[1].substring( j[1].search("name")+5,j[1].length());
+			start = nameField.indexOf("\"")+1;
+			end = nameField.indexOf("\"",start);
+			names[k] = nameField.substring(start,end);
+			for(ii=2;ii<j.length-1;ii++){
+				if(j[ii].length() == 0) //find a blank line
+					break;
+			}
+			if(ii == j.length-1)
+				values[k] = "";
```

Comment:
This statement can be extract from both if/else blocks (allowing to remove the "else" block).